sparse/src/KokkosSparse_spgemm_handle.hpp

```diff
@@ -639,77 +675,92 @@ class SPGEMMHandle {
   }
 #endif
 
-  void choose_default_algorithm() {
+  static
+  SPGEMMAlgorithm 
+  default_spgemm_algorithm() {
```

Comment:
This is part of Kokkos Kernels public API, you are not allowed to change it.
If you need to the new signature you can add it to the API and if for some reason you need to remove the old one, it needs to be marked as deprecated and should preferably be implemented by calling the new interface.